tangelo/linq/target/target_qiskit.py

```diff
@@ -92,11 +96,34 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
             job_sim = self.qiskit.execute(translated_circuit, backend, noise_model=qiskit_noise_model,
                                           shots=self.n_shots, basis_gates=None, optimization_level=opt_level)
             sim_results = job_sim.result()
-            frequencies = {state[::-1]: count/self.n_shots for state, count in sim_results.get_counts(0).items()}
+            self.all_frequencies = {state[::-1]: count/self.n_shots for state, count in sim_results.get_counts(0).items()}
 
-            self.all_frequencies = frequencies.copy()
+            frequencies = self.all_frequencies
             self._current_state = None
 
+        # desired_meas_result is not None and return_statevector is requested so loop shot by shot (much slower)
+        elif desired_meas_result is not None:
+            from tangelo.linq.noisy_simulation.noise_models import get_qiskit_noise_model
+            n_meas = source_circuit.counts.get("MEASURE", 0)
+            backend = self.AerSimulator(method='statevector')
+            qiskit_noise_model = get_qiskit_noise_model(self._noise_model) if self._noise_model else None
+            translated_circuit = self.qiskit.transpile(translated_circuit, backend)
+            translated_circuit.save_statevector()
+            samples = dict()
+            self._current_state = None
+
+            for _ in range(self.n_shots):
+                sim_results = backend.run(translated_circuit, noise_model=qiskit_noise_model, shots=1).result()
+                current_state = sim_results.get_statevector(translated_circuit)
+                measure = next(iter(self.qiskit.result.marginal_counts(sim_results, indices=list(range(n_meas))).get_counts()))[::-1]
+                (sample, _) = self.qiskit.quantum_info.states.Statevector(current_state).measure()
+                bitstr = measure + sample[::-1]
+                samples[bitstr] = samples.get(bitstr, 0) + 1
+                if measure == desired_meas_result or self._current_state is None:
```

Comment:
I do not understand this either (see similar comment above).
I'm not sure why L105 is useful as a result.